core/src/main/java/com/netflix/conductor/core/execution/mapper/KafkaPublishTaskMapper.java

```diff
@@ -46,14 +46,10 @@ public KafkaPublishTaskMapper(ParametersUtils parametersUtils, MetadataDAO metad
 		String taskId = taskMapperContext.getTaskId();
 		int retryCount = taskMapperContext.getRetryCount();
 
-		TaskDef taskDefinition = Optional.ofNullable(taskMapperContext.getTaskDefinition())
-				.orElseGet(() -> Optional.ofNullable(metadataDAO.getTaskDef(taskToSchedule.getName()))
-						.orElseThrow(() -> {
-							String reason = String.format("Invalid task specified. Cannot find task by name %s in the task definitions", taskToSchedule.getName());
-							return new TerminateWorkflowException(reason);
-						}));
+		Optional<TaskDef> taskDefinition = Optional.fromNullable(taskMapperContext.getTaskDefinition())
```

Comment:
Consider not defining taskDefinition as `Optional` from the guava library. Please use `Optional` from `java.util` package. Also `taskDefinition` here does not need to be an optional, since null value is passed as a param for it in line 52.